### DISCUSSION TITLE
Need advice on how to install pnpm

### DISCUSSION THREAD

**Discussion by JunkyardCat** [pnpm/pnpm#7073] [Q&A]
im using latest debian bookworm and have trouble installing. i installed it using npx 
npx pnpm setup
npx pnpm i -g pnpm
but still experience issue on the second one. `ERR_PNPM_NO_GLOBAL_BIN_DIR Unable to find the global bin directory`

then i tried npm i -g pnpm but also got an error `error: EACCES: permission denied, mkdir '/usr/lib/node_modules/pnpm'`

pls can someone explain why this is happening and how i can solve this

**Comment by JunkyardCat** [comment]
anyone can help me?

**Comment by JunkyardCat** [comment]
anyone?

**Comment by Shinyaigeek** [comment]
https://docs.npmjs.com/resolving-eacces-permissions-errors-when-installing-packages-globally Maybe this document will help you to tackle EACCESS error

**Reply by JunkyardCat** [reply]
is installing pnpm through npm the best choice for me or should i use the script
`curl -fsSL https://get.pnpm.io/install.sh | sh -`
also what does posix mean?

when i used the script this happened

![image](https://github.com/pnpm/pnpm/assets/22941307/d869afff-9164-435e-a811-d5ac425236a7)

**Comment by nurturenature** [comment]
Hi,

I was searching the same question.
Here's what resolved it for me on Debian Bookworm.

Note, node is already installed.

```bash
npm install pnpm
```

Which updated `package,json`
```json
"pnpm": "^9.1.1"
```
